pkg/variables/parsers/VariableTemplateParser.go

```diff
@@ -21,18 +22,102 @@ type VariableTemplateParser interface {
 	ExtractVariables(template string) ([]string, error)
 	//ParseTemplate(template string, values map[string]string) string
 	ParseTemplate(parserRequest VariableParserRequest) VariableParserResponse
+	//ParseTemplateWithPrimitiveHandling(parserRequest VariableParserRequest) VariableParserResponse
 }
 
 type VariableTemplateParserImpl struct {
-	logger *zap.SugaredLogger
+	logger                       *zap.SugaredLogger
+	VariableTemplateParserConfig *VariableTemplateParserConfig
 }
 
-func NewVariableTemplateParserImpl(logger *zap.SugaredLogger) *VariableTemplateParserImpl {
-	return &VariableTemplateParserImpl{logger: logger}
+func NewVariableTemplateParserImpl(logger *zap.SugaredLogger) (*VariableTemplateParserImpl, error) {
+	impl := &VariableTemplateParserImpl{logger: logger}
+	cfg, err := GetVariableTemplateParserConfig()
+	if err != nil {
+		return nil, err
+	}
+	impl.VariableTemplateParserConfig = cfg
+	return impl, nil
+}
+
+type VariableTemplateParserConfig struct {
+	ScopedVariableEnabled          bool `env:"SCOPED_VARIABLE_ENABLED" envDefault:"false"`
+	ScopedVariableHandlePrimitives bool `env:"SCOPED_VARIABLE_HANDLE_PRIMITIVES" envDefault:"false"`
+}
+
+func GetVariableTemplateParserConfig() (*VariableTemplateParserConfig, error) {
+	cfg := &VariableTemplateParserConfig{}
+	err := env.Parse(cfg)
+	return cfg, err
+}
+
+func preProcessPlaceholder(template string) string {
+	pattern := `\"@{{([^}]+)}}\"`
+	//@\{\{[a-zA-Z0-9-+/*%_\s]+\}\}
+
+	re := regexp.MustCompile(pattern)
+	matches := re.FindAllStringSubmatch(template, -1)
+
+	// Replace the surrounding quotes
+	for _, match := range matches {
+		if len(match) == 2 {
+			originalMatch := match[0]
+			innerContent := match[1]
+			replacement := fmt.Sprintf("@{{%s}}", innerContent)
+			template = strings.Replace(template, originalMatch, replacement, 1)
+		}
+	}
+	return template
+}
+
+func postProcessPlaceholder(template string) string {
+	pattern := `@{{([^}]+)}}`
+
+	re := regexp.MustCompile(pattern)
+	matches := re.FindAllStringSubmatch(template, -1)
+
+	// Replace the surrounding quotes
+	for _, match := range matches {
+		if len(match) == 2 {
+			originalMatch := match[0]
+			innerContent := match[1]
+			replacement := "\"" + innerContent + "\""
+			template = strings.Replace(template, originalMatch, replacement, 1)
+		}
+	}
+	return template
+}
```

Comment:
these two methods can be merged

Reply:
have removed post-processor in the new logic